spec/controllers/line_items_controller_spec.rb

```diff
@@ -84,6 +84,19 @@
                 expect(response.status).to eq 204
                 expect { item.reload }.to raise_error ActiveRecord::RecordNotFound
               end
+
+              context "after a payment is captured" do
+                let(:payment) { create(:check_payment, amount: order.total, order: order, state: 'completed') }
+                before { payment.capture! }
+
+                it 'updates the payment state' do
+                  expect(order.payment_state).to eq 'paid'
+                  delete :destroy, params
+                  expect(response.status).to eq 204
+                  order.update!
```

Comment:
we shouldn't need to update it in the test. That's something that deleting a line item should guarantee. I bet `order.reload` is enough. This will re-fetch the order from DB.

Reply:
Yes, it works with `order.reload` as well.

Maybe line 95 (checking the response status 204), could be deleted as well? This is actually checked in the previous test case.

Reply:
Agree. This test only cares about the assertion below and the redundancy has its cost.